device/dhcp/common/dhcpv6options.yaml

```diff
@@ -0,0 +1,387 @@
+            vendor_information:
+              x-field-uid: 1
+            name_servers:
+              x-field-uid: 2
+            fqdn:
+              x-field-uid: 3
+            bootfile_url:
+              x-field-uid: 4
+            sztp:
+              x-field-uid: 5
+            custom:
+              x-field-uid: 6
+        custom:
+          $ref: '#/components/schemas/Dhcpv6Options.Custom'
+          x-field-uid: 2
+
+    Dhcpv6Options.Custom:
+      description: >-
+        The Custom option is used to provide a not well known option in the message between a client and a server.
+      type: object
+      properties:
+        type:
+          description: >-
+            The type of the Custom option TLV.
+          type: integer
+          format: uint32
+          x-field-uid: 1  
+        value:
+          description: >-
+            The value of the Custom option TLV.
+          type: string
+          x-field-uid: 2
+
+    Dhcpv6Options.VendorClass:
+      description: >-
+        This option is used by a client to identify the vendor that manufactured the hardware on which the client is running.
+      type: object
+      properties:
+        enterprise_number:
+          description: >-
+            The vendor's registered Enterprise Number as registered with IANA.
+          type: integer
+          format: uint32
+          x-field-uid: 1
+        class_data: 
+          description: >-
+            The opaque data representing the hardware configuration of the host on which the client is running.
+          type: string
+          x-field-uid: 2
+
+    Dhcpv6Options.VendorInfo:
+      description: >-
+        This option is used by clients and servers to exchange vendor-specific information.
+      type: object
+      properties:
+        enterprise_number:
+          description: >-
+            The vendor's registered Enterprise Number as registered with IANA.
+          type: integer
+          format: uint32
+          x-field-uid: 1
+        option_data: 
+          description: >-
+            An opaque object of option-len octets,interpreted by vendor-specific code on the clients and servers.
+          type: array
+          items:
+            $ref: '#/components/schemas/Dhcpv6Options.VendorSpecificOptions'
+          x-field-uid: 2
+
+    Dhcpv6Options.VendorSpecificOptions:
+      description: >-
+        The encapsulated vendor-specific options field is encoded as a sequence of code/length/value fields of 
+        identical format to the DHCP options field. The option codes are defined by the vendor identified in the 
+        enterprise-number field and are not managed by IANA.
+      type: object
+      properties:
+        code:
+          description: >-
+            The code for the encapsulated option.
+          type: integer
+          format: uint32
+          x-field-uid: 1  
+        data:
+          description: >-
+            The data for the encapsulated option.
+          type: string
```

Comment:
Is hex a better choice ?

Reply:
vendor specific so kept as string

Reply:
Are these ascii characters ? Otherwise how will user specify non-printable characters. Can you maybe try adding a snippet exercising this attribute in the gosnappi example?